tests/unit/bin/test_acoustic.py

```diff
@@ -33,6 +33,23 @@ def test_main_force_resync_by_email_list(dbsession, maximal_contact, tmpdir):
     assert len(dbsession.query(PendingAcousticRecord).all()) > 0
 
 
+def test_main_force_resync_will_reset_existing_retries(
```

Comment:
Based on my [comment here](https://github.com/mozilla-it/ctms-api/issues/806#issuecomment-1716282544), should we add a test that demonstrates when there are two pending records for the same `email_id`, after running `do_sync`, only one will remain with `retry=0`?

Or we could land the fix for duplicate email_ids in the queue in this PR.